src/itest/kotlin/nl/lifely/zac/itest/SmartDocumentsTest.kt

```diff
@@ -0,0 +1,174 @@
+                    shouldContainJsonKeyValue("$[0].groups[0].templates[0].id", "7B7857BB9959470C82974037304E433D")
+                    shouldContainJsonKeyValue("$[0].groups[0].templates[0].name", "Data Test")
+                    shouldContainJsonKeyValue("$[0].groups[0].templates[1].id", "273C2707E5A844699B653C87ACFD618E")
+                    shouldContainJsonKeyValue("$[0].groups[0].templates[1].name", "OpenZaakTest")
+
+                    shouldContainJsonKeyValue("$[0].groups[1].id", "348097107FA346DC9AEBBE33A5500B79")
+                    shouldContainJsonKeyValue("$[0].groups[1].name", "OpenZaak")
+                    shouldContainJsonKeyValue("$[0].groups[1].templates[0].id", "7B7857BB9959470C82974037304E433D")
+                    shouldContainJsonKeyValue("$[0].groups[1].templates[0].name", "Data Test")
+                    shouldContainJsonKeyValue("$[0].groups[1].templates[1].id", "273C2707E5A844699B653C87ACFD618E")
+                    shouldContainJsonKeyValue("$[0].groups[1].templates[1].name", "OpenZaakTest")
+
+                    shouldContainJsonKeyValue("$[0].groups[2].id", "CC2696396E1D4899A74AC025B19C8FDC")
+                    shouldContainJsonKeyValue(
+                        "$[0].groups[2].name",
+                        "Indienen aansprakelijkstelling door derden behandelen"
+                    )
+                    shouldContainJsonKeyValue("$[0].groups[2].templates[0].id", "7B7857BB9959470C82974037304E433D")
+                    shouldContainJsonKeyValue("$[0].groups[2].templates[0].name", "Data Test")
+                    shouldContainJsonKeyValue("$[0].groups[2].templates[1].id", "273C2707E5A844699B653C87ACFD618E")
+                    shouldContainJsonKeyValue("$[0].groups[2].templates[1].name", "OpenZaakTest")
+
+                    shouldContainJsonKeyValue("$[0].templates[0].id", "2F426D9B06034FDB93FDC2C6427640DD")
+                    shouldContainJsonKeyValue("$[0].templates[0].name", "Aanvullende informatie nieuw")
+                    shouldContainJsonKeyValue("$[0].templates[1].id", "CDC0FD0F08B545B59E36D610AFFA0B58")
+                    shouldContainJsonKeyValue("$[0].templates[1].name", "Aanvullende informatie oud")
+                    shouldContainJsonKeyValue("$[0].templates[2].id", "40A8482A34F941618B95359BF8246FB8")
+                    shouldContainJsonKeyValue("$[0].templates[2].name", "Advies nieuw")
+                    shouldContainJsonKeyValue("$[0].templates[3].id", "16B8B5A8E18A4BA484302029B51C32B6")
+                    shouldContainJsonKeyValue("$[0].templates[3].name", "Advies oud")
+                    shouldContainJsonKeyValue("$[0].templates[4].id", "53AFB00EE74E4B8AB1EC73CCB3251F8B")
+                    shouldContainJsonKeyValue("$[0].templates[4].name", "Besluit nieuw")
+                    shouldContainJsonKeyValue("$[0].templates[5].id", "60D711D768184F7F94A8BC1553A689A1")
```

Comment:
Same here. To me testing so many templates here does not add any value?

Reply:
We convert from SD response to our REST model. 
So the mocked response is SD format and the test checks if the REST response has all the fields, as expected.

Reply:
Left only 2 templates to exercise loops & recursion